tinyiiod-private.h

```diff
@@ -41,9 +41,13 @@ void tinyiiod_do_write_attr(struct tinyiiod *iiod, const char *device,
 void tinyiiod_do_open(struct tinyiiod *iiod, const char *device,
 		      size_t sample_size, uint32_t mask);
 void tinyiiod_do_close(struct tinyiiod *iiod, const char *device);
+int32_t tinyiiod_do_close_instance(struct tinyiiod *iiod);
 
-void tinyiiod_do_readbuf(struct tinyiiod *iiod,
-			 const char *device, size_t bytes_count);
+int32_t tinyiiod_do_readbuf(struct tinyiiod *iiod,
+			    const char *device, size_t bytes_count);
+
+int32_t tinyiiod_do_writebuf(struct tinyiiod *iiod, const char *device,
+			     size_t bytes_count);
```

Comment:
From my understanding, this commit implements some new functions, such as` tinyiiod_do_writebuf() ` and modifies the implementation of already existing functions, such as `tinyiiod_do_readbuf()`. This is fine.
However, it looks like it also tries to change the return type of some functions, such as `tinyiiod_read()`. I would add these changes in a different commit. It makes the patch cleaner and easier to follow.